### DISCUSSION TITLE
Does OTEL Logs support the notion of the structured_metadata of Loki?

### DISCUSSION THREAD

**Discussion by trajano** [open-telemetry/community#1786] [Community Discussions]
I know at present that it doesn't have it at least based on my experiments.  Even with promtail it is a matter of extracting the data and placing them somewhere else.  Just wondering if it is in the roadmap?

The use of [structured_metadata](https://grafana.com/docs/loki/latest/send-data/promtail/stages/structured_metadata/) is experimental in Loki and it is a to work around a technical implementation where it is inefficient to label high cardinality items such as trace_id, user IDs, access_tokens but we still want to be able to view them separately from the log message.

Secondly if we had something like this we can have a common structure like

```json
{
  "message": "This is the message",
  "log.original": "message=\"This is the message\" info=\"ignored 123\" user_id=456 trace_id=asdfasdfasdf",
  "metadata": {
    "otel.traceid": "asdfasdfasdf"
    "user_id": 456
  },
  "attributes": {
    "stream": "stderr",
    "service.name": "my service"
  }
}
```

**Comment by tigrannajaryan** [comment]
Are you using Collector with [lokiexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/lokiexporter)? If so then best to ask this question in Collector [discussions](https://github.com/open-telemetry/opentelemetry-collector-contrib/discussions).

**Reply by trajano** [reply]
I put it here because I was thinking do the OTEL protocols actually have (or planned) support for structured metadata?  Not specific to Loki.  And the lokiexporter itself doesn't support it, but I don't know if it is a gap on the exporter or a gap in the standard.

**Reply by tigrannajaryan** [reply]
I am not sure what you mean by "metadata" but yes OTLP (OpenTelemetry's protocol) indeed supports structured data in the log record's Body and Attributes. It can be seen in the LogRecord Protobuf definition: https://github.com/open-telemetry/opentelemetry-proto/blob/ea449ae0e9b282f96ec12a09e796dbb3d390ed4f/opentelemetry/proto/logs/v1/logs.proto#L134

**Reply by trask** [reply]
hi @trajano, since you've been involved in the Java repos, I'm guessing you may be interested in the recent work there to support structured log bodies:
* https://github.com/open-telemetry/opentelemetry-java/pull/5880
* https://github.com/open-telemetry/opentelemetry-java/pull/5938

**Reply by trajano** [reply]
Yup I see it supports structured data but with Loki and I presume others making map to their respective index.  However, I am thinking there may need to be a separation of high-cardinality items (you have it for `trace_id` already) but other things that may not be general enough for OTEL like `username`. 

I'm marking @tigrannajaryan  comment as the answer since it seems the standard does support it, and it would just be up to lokireceiver and lokiexporter to handle it.
